File: killrvideo-service-search/src/main/java/com/killrvideo/service/search/request/SearchVideosRequestData.java
package com.killrvideo.service.search.request;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.ToString;

import java.util.Optional;

@Getter @AllArgsConstructor @ToString
public class SearchVideosRequestData {
    private String query;
    private int pageSize;
    @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
    private Optional<String> pagingState;
}

File: killrvideo-service-search/src/main/kotlin/com/killrvideo/service/search/request/SearchVideosRequestData.kt
package com.killrvideo.service.search.request;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.ToString;

import java.util.Optional;

@Getter @AllArgsConstructor @ToString
public class SearchVideosRequestData {
    private final String query;
    private final int pageSize;
    @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
    private final Optional<String> pagingState;
}
